lib/transformer/caster.rb

```diff
@@ -5,39 +5,41 @@ module Caster
     #
     # Arguments:
     # - dataset       = Atlas::Dataset
-    # - dataset_edits = Hash[<key> => Float]
+    # - dataset_cast = Hash[<key> => Float]
     #
-    def self.cast(dataset, dataset_edits)
-      validator = AttributeValidator.new(dataset_edits)
+    def self.cast(dataset, dataset_cast_attributes)
+      dataset_cast = DatasetCast.new(filter(dataset_cast_attributes))
 
-      unless validator.valid?
-        raise ArgumentError, validator.message
+      unless dataset_cast.valid?
+        raise ArgumentError, dataset_cast.errors.full_messages.join(', ')
       end
 
-      casts = [
-        Caster::ElectricityConsumption,
-        Caster::Lighting,
-        Caster::Appliances,
-        Caster::GasConsumption,
-        Caster::Cooking,
-        Caster::Cooling,
-        Caster::WaterHeater,
-        Caster::SpaceHeater,
-        Caster::RoofSurfaceAvailableForPV,
-        Caster::Transport,
-        Caster::Buildings,
-        Caster::Industry,
-        Caster::Agriculture,
-        Caster::EnergySector,
-        Caster::NullAttributes,
-        Caster::ToAtlasAttribute
-      ].freeze
+      Caster::ToAtlasAttribute
+        .filter(dataset_cast, casts.reduce(Template.new) do |object, analyzer|
+          analyzer.analyze(dataset_cast, object)
+        end)
+    end
 
-      graph = Atlas::Runner.new(dataset).calculate
+    private
 
-      casts.reduce({}) do |object, analyzer|
-        (casts.last == analyzer ? {} : object).merge(analyzer.analyze(dataset, graph, dataset_edits.symbolize_keys, object))
-      end
+    def self.filter(dataset_cast_attributes)
+      dataset_cast_attributes
```

Comment:
@antw I love the `graph_methods` as an `OpenStruct`. But the attributes from ETlocal do arrive in a single Hash (so no splitting into graph methods and area attributes), is it okay to do this in this Class or should ETlocal be the application to do this properly? I have no strong feelings but maybe something you say make me feel otherwise.

Reply:
> ETlocal do arrive in a single Hash

I'm curious why. Is there an advantage to keeping one big hash in ETLocal instead of separate structures for graph methods and area data?

I've no strong opinion either, so if you think this is for the best then I'm perfectly happy with it.